internal/provider/cluster_resource_test.go

```diff
@@ -757,6 +785,20 @@ func testDedicatedClusterResource(
 			// turned off due to this reason.
 			ImportStateVerify: false,
 		},
+		{
+			Config: getTestDedicatedClusterResourceConfig(clusterName, latestClusterMajorVersion, false, 2, ptr(true)),
+			Check:  resource.TestCheckResourceAttr(resourceName, "delete_protection", "true"),
+		},
+		{
+			// Delete step that fails since delete protection is enabled.
+			Config:      " ",
+			Destroy:     true,
+			ExpectError: regexp.MustCompile(".*Cannot destroy cluster with delete protection enabled*"),
```

Comment:
nit: It might be nice to show that resources that are attached to cluster have not been destroyed.   Sorry I keep finding one off things.  This one is optional.

Reply:
Talked about this offline. There aren't already any peripheral resources for this cluster and it would be painful and make the test significantly more complicated to add them so I'll skip it for now.